lib/ozfarium_web/live/gallery.ex

```diff
@@ -186,38 +186,15 @@ defmodule OzfariumWeb.Live.Gallery do
 
     {:noreply, after_visibility_update(socket, ozfa)}
   end
-
- @impl true
-def handle_event("increment-ep-count", %{"id" => id}, %{assigns: assigns} = socket) do
-  ozfa = Gallery.get_ozfa!(id)
-  user_ozfa = Gallery.find_user_ozfa(ozfa, assigns.current_user)
-
-  current_ep_timestamps = user_ozfa.ep_timestamps || []
-
-  updated_ep_timestamps = [
-    NaiveDateTime.utc_now() | current_ep_timestamps
-  ]
-
-  Gallery.update_user_ozfa(user_ozfa, %{ep_timestamps: updated_ep_timestamps})
-  {:noreply, after_visibility_update(socket, ozfa)}
-end
-
   
   @impl true
-  def handle_event("decrement-ep-count", %{"id" => id}, %{assigns: assigns} = socket) do
+  def handle_event("count-ep", %{"id" => id, "action" => action}, %{assigns: assigns} = socket) do
     ozfa = Gallery.get_ozfa!(id)
-    user_ozfa = Gallery.find_user_ozfa(ozfa, assigns.current_user)
-    current_ep_timestamps = user_ozfa.ep_timestamps
-
-    if Enum.any?(current_ep_timestamps) do
-      updated_ep_timestamps = Enum.drop(current_ep_timestamps, 1)
-      Gallery.update_user_ozfa(user_ozfa, %{ep_timestamps: updated_ep_timestamps})
-    end
-
+    user = assigns.current_user
+    Gallery.count_ep(ozfa.id, user, action)
     {:noreply, after_visibility_update(socket, ozfa)}
```

Comment:
Thinking about it, there is no visibility update, no need to re-fetch the whole list.
This should be enough:

```
  def handle_event("count-ep", %{"id" => id, "action" => action}, %{assigns: assigns} = socket) do
    case Gallery.count_ep(ozfa.id, assigns.current_user, action) do
     {:ok, _} ->
        {:noreply, assign(socket, ozfa: Gallery.preload_ozfa!(assigns.current_user, ozfa.id))}

     {:error, changeset} ->
        {:noreply, socket}
     end
  end
```

Reply:
done